aiverify-foundation/moonshot
# [MS-427] Add persistent volume to data

## Description

Currently, when EKS instance crashes, all endpoints gets deleted and/or resetting to default settings and benchmark runs and sessions gets deleted.

## Motivation and Context

Currently, when EKS instance crashes, all endpoints gets deleted and/or resetting to default settings and benchmark runs and sessions gets deleted.

With this change, there will be a persistent volume created where the data will be stored.

## Type of Change


- feat: A new feature










## How to Test

EKS Website
`https://moonshot.eks-wb.aipo-imda.net/`

Assume role:
`. ./assume-wb-role.sh `

Get the running pod:
`kubectl get pod -n moonshot`

Kill the running pod:
`kubectl delete pod moonshot-ui-deployment-6cb446bbc5-lbk5g -n moonshot`

See from the UI that it is dead, then get the running pod to see that the old is terminated and the new one is in effect.
Look at the files and can see that it is still working

Use logs to check the current status:
Note the following logs:

2024-10-10 04:53:59,396 [INFO][__main__.py::moonshot_data_installation(131)] Installing Moonshot Data from GitHub

2024-10-10 04:53:59,396 [WARNING][__main__.py::moonshot_data_installation(141)] Directory moonshot-data already exists.

**2024-10-10 04:53:59,396 [INFO][__main__.py::moonshot_data_installation(158)] Unattended mode detected. To reinstall, please use the --overwrite flag.**

2024-10-10 04:53:59,396 [INFO][__main__.py::moonshot_data_installation(173)] Installing requirements for moonshot-data

`kubectl logs moonshot-ui-deployment-58b86c7666-hzwzp -c moonshot-ui -n moonshot`

## Checklist

Please check all the boxes that apply to this pull request using "x":


## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>
